test/api/options.js

```diff
@@ -289,4 +289,66 @@ describe("API: constructor options", () => {
       });
     });
   });
+
+  describe("storageQuota", () => {
+    describe("not set", () => {
+      it("should be 5000000 bytes by default", () => {
+        const { localStorage, sessionStorage } = (new JSDOM(``, { url: "https://example.com" })).window;
+        const dataWithinQuota = "0".repeat(4000000);
+
+        localStorage.setItem("foo", dataWithinQuota);
+        sessionStorage.setItem("bar", dataWithinQuota);
+
+        assert.strictEqual(localStorage.foo, dataWithinQuota);
+        assert.strictEqual(sessionStorage.bar, dataWithinQuota);
+
+        const dataExceedingQuota = "0".repeat(6000000);
+
+        assert.throws(() => localStorage.setItem("foo", dataExceedingQuota));
+        assert.throws(() => sessionStorage.setItem("bar", dataExceedingQuota));
+      });
+    });
+
+    describe("set to 10000 bytes", () => {
+      it("should only allow setting data within the custom quota", () => {
+        const { localStorage, sessionStorage } = (new JSDOM(``, {
+          url: "https://example.com",
+          storageQuota: 10000
+        })).window;
+        const dataWithinQuota = "0".repeat(5);
+
+        localStorage.setItem("foo", dataWithinQuota);
+        sessionStorage.setItem("bar", dataWithinQuota);
+
+        assert.strictEqual(localStorage.foo, dataWithinQuota);
+        assert.strictEqual(sessionStorage.bar, dataWithinQuota);
+
+        const dataExceedingQuota = "0".repeat(15000);
```

Comment:
It would be interesting to test something like 9995 here, since even though the string itself won't exceed the quota, the sum of all strings will.

Reply:
Thanks for the suggestion! I'm not sure if this is what you were thinking of, but it revealed an issue where replacing the value of an existing key would include the size of both entries in the count. I've fixed this and added the test now.